sktime/forecasting/rbf_forecaster.py

```diff
@@ -0,0 +1,523 @@
+"""RBF Neural Networks for Time Series Forecasting."""
+
+__author__ = ["phoeenniixx"]
+
+import numpy as np
+import pandas as pd
+from sklearn.preprocessing import StandardScaler
+
+from sktime.forecasting.base.adapters._pytorch import BaseDeepNetworkPyTorch
+from sktime.utils.dependencies._dependencies import _check_soft_dependencies
+from sktime.utils.warnings import warn
+
+if _check_soft_dependencies("torch", severity="warning"):
+    import torch
+    import torch.nn as nn
+
+else:
+
+    class nn:
+        """dummy class if torch is not available."""
+
+        class Module:
+            """dummy class if torch is not available."""
+
+            def __init__(self, *args, **kwargs):
+                raise ImportError("torch is not available. Please install torch first.")
+
+
+class RBFLayer(nn.Module):
+    r"""RBF layer to transform input data into a new feature space.
+
+    This layer applies an RBF transformation to each input feature,
+    expanding the feature space based on distances to predefined center points.
+
+    Parameters
+    ----------
+    in_features : int
+        Number of input features
+    out_features : int
+        Number of output features (number of RBF centers)
+    centers : torch.Tensor, optional (default=None)
+        The centers :math:`c_k` for the RBF transformation.
+        If None, centers are evenly spaced.
+    gamma : float, optional (default=1.0)
+        Parameter controlling the spread of the RBFs
+    rbf_type : {"gaussian", "multiquadric", "inverse_multiquadric"},
+        optional (default="gaussian")
+        The type of RBF kernel to apply.
+        - "gaussian": :math:`\\exp(-\\gamma (t - c)^2)`
+        - "multiquadric": :math:`\\sqrt{1 + \\gamma (t - c)^2}`
+        - "inverse_multiquadric": :math:`\frac{1}{\\sqrt{1 + \\gamma (t - c)^2}}`
+
+    """
+
+    def __init__(
+        self, in_features, out_features, centers=None, gamma=1.0, rbf_type="gaussian"
+    ):
+        super().__init__()
+        self.in_features = in_features
+        self.out_features = out_features
+        self.gamma = gamma
+        self.rbf_type = rbf_type.lower()
+
+        if centers is None:
+            centers = torch.linspace(-1, 1, out_features).reshape(-1, 1)
+            centers = centers.repeat(1, in_features)
+        else:
+            centers = torch.as_tensor(centers, dtype=torch.float32)
+
+        self.centers = nn.Parameter(centers, requires_grad=True)
+
+        valid_rbf_types = {"gaussian", "multiquadric", "inverse_multiquadric"}
+        if self.rbf_type not in valid_rbf_types:
+            raise ValueError(
+                f"rbf_type must be one of {valid_rbf_types}, got {self.rbf_type}"
+            )
+
+    def forward(self, x):
```

Comment:
Perhaps you can reduce the code duplication in RBFTransformer by importing this Layer in case torch should be used by RBFTransformer. (Might be a separate PR).

Reply:
Would appreciate an Issue, after the pr is merged.